pandas/core/dtypes/cast.py

```diff
@@ -2062,8 +2062,17 @@ def maybe_cast_to_integer_array(
     # doesn't handle `uint64` correctly.
     arr = np.asarray(arr)
 
-    if is_unsigned_integer_dtype(dtype) and (arr < 0).any():
-        raise OverflowError("Trying to coerce negative values to unsigned integers")
+    if is_unsigned_integer_dtype(dtype):
+        try:
+            if (arr < 0).any():
```

Comment:
why is this is a try except

Reply:
The input in arr is ["0", "1", "2"], this is causing a TypeError in the check ```(arr < 0).any()``` but it can be casted to uint, so I am checking ```(casted < 0).any()``` which it hits and casts correctly

Reply:
this doesn't make sense
why does this check matter? is the casted valid? 

where is the test that checks the overflow?

strive for minimal code

Reply:
```result = Series(["0", "1", "2"], dtype=uint8)```, this should give a valid cast and casted is valid, but ```(arr < 0).any()``` is giving TypeError.

Reply:
The point is that string arrays cannot be compared with integers, numpy does not support it, thus it is leading to a Type Error.